### PR TITLE
Fallback to regular if cog base image can't be determined and flag is not explicitly set

### PR DESCRIPTION
**Pull request by 8W9aG**
* Add base-image to gitignore
* Use a boolean pointer for cog base image flag for a tri-state variable
* Handle not explicitly setting the cog base image and a failure to resolve the base image as a warning
* Add integration test for torch 1.13.0

* Solves problem 3 in: https://github.com/replicate/cog/issues/1862

### CODE REVIEW

File: pkg/cli/build.go [hunk 1]
```diff
@@ -116,7 +112,11 @@ func addDockerfileFlag(cmd *cobra.Command) {
 }
 
 func addUseCogBaseImageFlag(cmd *cobra.Command) {
-	cmd.Flags().BoolVar(&buildUseCogBaseImage, "use-cog-base-image", true, "Use pre-built Cog base image for faster cold boots")
+	buildUseCogBaseImage = new(bool)
+	cmd.Flags().BoolVar(buildUseCogBaseImage, useCogBaseImageFlagKey, true, "Use pre-built Cog base image for faster cold boots")
+	if !cmd.Flags().Changed(useCogBaseImageFlagKey) {
+		buildUseCogBaseImage = nil
+	}
```
Comment (mattt) [inline pkg/cli/build.go:119]: Wait, does this work? My mental model of `cmd.Flags().Changed()` was that it evaluated in `RunE` when args were passed in, like a variable initialized and then set later.

**Review by mattt — APPROVED** [review]

File: pkg/cli/build.go (hunk 1 above)
Comment (8W9aG) [reply pkg/cli/build.go:119]: You are completely correct, let me find a different way to express this.

Comment (8W9aG) [reply pkg/cli/build.go]: I've got a fix and added some more tests, let me know if this is sufficient.

File: test-integration/test_integration/test_build.py [hunk 2]
```diff
@@ -240,3 +240,49 @@ def test_build_base_image_sha(docker_image):
...
+            "-t",
+            docker_image,
+            "--openapi-schema",
+            "openapi.json",
+            "--use-cog-base-image=false",
+        ],
+        cwd=project_dir,
+        capture_output=True,
+    )
+    assert build_process.returncode == 0
```
Comment (mattt) [inline test-integration/test_integration/test_build.py:288]: Lovely to have all of these. Thank you.

**Review by mattt — APPROVED** [review]

**Pull request merged**